scripts/cbmc_json_parser.py

```diff
@@ -0,0 +1,400 @@
+    result = subprocess.run(command, capture_output=True,universal_newlines=True)
+    json_file = json.loads(result.stdout)
+    print(json_file)
+    return
+
+def transform_cbmc_output(log_file, output_style):
+    """
+    Take Unstructured CBMC Response object, parse the blob and gives structured 
+    and formatted output depending on User Inputted Output Style
+
+    Parameters -
+        log_file : str
+            A response blob that is given to the function from CBMC
+        output_style : int
+            An index to tell the script which style of output is requested by the User.
+
+    Returns - 
+        None
+            Prints the final output string to the user
+    """
+
+    # Enum to store the style of output that is given by the argument flags
+    output_style_switcher = {
+        '0' : 'default',
+        '1' : 'terse',
+        '2' : 'old'
+    }
+
+    # Output Message is the final output that is printed to the user
+    output_message = ""
+
+    # Check if the output given by CBMC is in Json format or not
+    if is_json(log_file):
+        json_file = json.loads(log_file)
+        
+        # Write cbmc output to a json file in scripts/temp_json for parallel I/O
+        json_file = save_json_file(json_file)
+        
+        # Read from the saved json file and restructure it to make it parseable
+        parseable_json_dictionary = restructure_json_file(json_file)
+        
+        # Extract property information from the restructured JSON file 
+        properties = extract_properties(parseable_json_dictionary)
+        
+        # Using Case Switching to Toggle between various output styles
+        # For now, the two options provided are default and terse
+        if output_style_switcher[output_style] == 'default':
+            
+            # Extract Solver Information from the json file and construct message 
+            solver_information = extract_solver_information(parseable_json_dictionary)
+            output_message += construct_solver_information_message(solver_information)
+
+            # Extract property messages from the json file
+            # Combine both messages to give as final output 
+            output_message += construct_property_message(properties)
+        
+        elif output_style_switcher[output_style] == 'terse':
+            
+            # Construct only summarized result and display output 
+            output_message = construct_terse_property_message(properties)
+        
+        # Print using an Interfact function
+        print_to_terminal(output_message)
+
+        # Delete temp Json file after displaying result
+        clear_json_file(json_file)
+    else:
+        # DynTrait tests generate a non json output due to "Invariant check failed" error
+        # For these cases, we just produce the cbmc output unparsed
+        # TODO: Parse these non json outputs from CBMC
+        non_json_cbmc_output_handler(log_file)
```

Comment:
The following files in the regression tests (all related to DynTraits) give non json cbmc output, so for now the handling is to just print out the direct output (for the regression tests). Raising an exception would fail the regression tests. I thought I could handle these as a seperate Issue.
```
rmc/DynTrait/drop_nested_boxed_dyn_fail.rs
rmc/DynTrait/dyn_fn_mut_fail.rs
 rmc/DynTrait/dyn_fn_once_fail.rs
rmc/DynTrait/nested_closures_fail.rs
```

Reply:
Can you add to these tests the option to use the old output format? You can set rmc flags by adding a comment at the top of the rust file:

```rust
// rmc-flags: <option>
```